### PR TITLE
[billing] Report metrics on updated subscriptions

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Reports metric on the number of usage records udpated in stripe. This can be used to observe trends and spot issues.

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

**Comment by roboquat** [comment]
@easyCZ: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/usage/pkg/stripe/stripe.go
Comment (andrew-farries) [inline components/usage/pkg/stripe/stripe.go:94]: The `*UsageRecord` return value is ignored by its only caller. is that intentional? If so can we remove the return value here?

**Review by andrew-farries — COMMENTED** [review]
Did you test Stripe invoice updating still works after the `updateUsageForCustomer` refactor?

We don't have automated tests for the integration with Stripe.

File: components/usage/pkg/stripe/stripe.go
Comment (easyCZ) [reply components/usage/pkg/stripe/stripe.go:94]: I was planning to add a second metric which shows total amount of usage increase as well as return it in the RPC. You're right, it's not needed in this PR directly. I can remove it if you want

Comment (andrew-farries) [reply components/usage/pkg/stripe/stripe.go:94]: I'll leave it up to you. No point in removing if the next PR puts it straight back.

**Review by andrew-farries — APPROVED** [review]
/hold for possible manual test of the Stripe integration.

**Comment by easyCZ** [comment]
/unhold

I'll make use of the field in subsequent PR.

**Pull request merged**
